src/page/product-list.page.ts

```diff
@@ -1,15 +1,28 @@
 import {
-  $, ElementFinder, ExpectedConditions, browser,
+  $, $$, ElementArrayFinder, ExpectedConditions, browser, ElementFinder,
 } from 'protractor';
 
 export class ProductListPage {
+  private product_list: ElementArrayFinder;
+
   private addProductMenu: ElementFinder;
 
   constructor() {
-    this.addProductMenu = $('#center_column [title="Add to cart"]');
+    this.product_list = $$('ul.product_list');
+    this.addProductMenu = $('[id="add_to_cart"]');
+  }
+
+  private findByProduct(productName: string): ElementFinder {
+    return this.product_list.filter((product: ElementFinder) => product.$('.product-name').getText().then((name: string) => name.includes(productName))).first();
```

Comment:
could you split this line to make it easier to read :)

Reply:
Done :)